### PR TITLE
193 full name

### PR DESCRIPTION
**Pull request by NCFindley**
Please review :).  Finish update on Full Name issue and remove first/last name columns.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -8,8 +8,7 @@ class ApplicationController < ActionController::Base
   protected
 
   def configure_permitted_parameters
-    devise_parameter_sanitizer.for(:sign_up) << :first_name
-    devise_parameter_sanitizer.for(:sign_up) << :last_name
+    devise_parameter_sanitizer.for(:sign_up) << :full_name
```
Comment (sumeetjain) [inline app/controllers/application_controller.rb:11]: Calls devise_parameter_sanitizer.for(:sign_up) 4 times (DuplicateMethodCall)

File: db/migrate/20160811194649_add_full_name_to_user.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require 'pry'
+class AddFullNameToUser < ActiveRecord::Migration
+  def change
+  	add_column :users, :full_name, :string
+
+  	 User.find_each do |user|
+	
+  		fullname = user.first_name + " " + user.last_name
```
Comment (sumeetjain) [inline db/migrate/20160811194649_add_full_name_to_user.rb]: Refers to user more than self (maybe move it to another class?) (FeatureEnvy)

File: db/migrate/20160811194649_add_full_name_to_user.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require 'pry'
+class AddFullNameToUser < ActiveRecord::Migration
```
Comment (sumeetjain) [inline db/migrate/20160811194649_add_full_name_to_user.rb]: Has no descriptive comment (IrresponsibleModule)

File: db/migrate/20160812144450_remove_first_last_column_in_users.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class RemoveFirstLastColumnInUsers < ActiveRecord::Migration
```
Comment (sumeetjain) [inline db/migrate/20160812144450_remove_first_last_column_in_users.rb]: Has no descriptive comment (IrresponsibleModule)

File: app/models/user.rb [hunk 5]
```diff
@@ -11,8 +11,7 @@ class User < ActiveRecord::Base
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
 
-  validates :first_name, presence: true
-  validates :last_name, presence: true
+  validates :full_name, presence: true
```
Comment (sumeetjain) [inline app/models/user.rb:14]: Inconsistent indentation detected.

File: app/models/user.rb [hunk 6]
```diff
@@ -134,7 +133,7 @@ def email_required?
   end
 
   def self.to_csv
-    attributes = %w[secret_number last_name first_name]
+    attributes = %w[secret_number full_name]
```
Comment (sumeetjain) [inline app/models/user.rb:136]: `%w`-literals should be delimited by `(` and `)`.

File: db/migrate/20160811194649_add_full_name_to_user.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  def change
+  	add_column :users, :full_name, :string
```
Comment (sumeetjain) [inline db/migrate/20160811194649_add_full_name_to_user.rb:4]: Use 2 (not 1) spaces for indentation.

Comment (sumeetjain) [inline db/migrate/20160811194649_add_full_name_to_user.rb:4]: Tab detected.

File: db/migrate/20160811194649_add_full_name_to_user.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+  	 User.find_each do |user|
```
Comment (sumeetjain) [inline db/migrate/20160811194649_add_full_name_to_user.rb:6]: Tab detected.

Comment (sumeetjain) [inline db/migrate/20160811194649_add_full_name_to_user.rb:6]: Inconsistent indentation detected.

File: db/migrate/20160811194649_add_full_name_to_user.rb [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+
```
Comment (sumeetjain) [inline db/migrate/20160811194649_add_full_name_to_user.rb:7]: Tab detected.

Comment (sumeetjain) [inline db/migrate/20160811194649_add_full_name_to_user.rb:7]: Trailing whitespace detected.

File: db/migrate/20160811194649_add_full_name_to_user.rb (hunk 2 above)
Comment (sumeetjain) [inline db/migrate/20160811194649_add_full_name_to_user.rb:8]: Tab detected.

Comment (sumeetjain) [inline db/migrate/20160811194649_add_full_name_to_user.rb:8]: Use 2 (not 0) spaces for indentation.

File: db/migrate/20160811194649_add_full_name_to_user.rb [hunk 10]
```diff
@@ -0,0 +1,13 @@
+require 'pry'
+class AddFullNameToUser < ActiveRecord::Migration
+  def change
+  	add_column :users, :full_name, :string
+
+  	 User.find_each do |user|
+	
+  		fullname = user.first_name + " " + user.last_name
+  		user.full_name = fullname
```
Comment (sumeetjain) [inline db/migrate/20160811194649_add_full_name_to_user.rb:9]: Tab detected.

File: db/migrate/20160811194649_add_full_name_to_user.rb [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+  		user.save
```
Comment (sumeetjain) [inline db/migrate/20160811194649_add_full_name_to_user.rb:10]: Tab detected.

File: db/migrate/20160811194649_add_full_name_to_user.rb [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+  	 end
```
Comment (sumeetjain) [inline db/migrate/20160811194649_add_full_name_to_user.rb:11]: Tab detected.

File: db/migrate/20160812144450_remove_first_last_column_in_users.rb [hunk 13]
```diff
@@ -0,0 +1,6 @@
...
+  def change
+  	remove_column :users, :first_name, :string
```
Comment (sumeetjain) [inline db/migrate/20160812144450_remove_first_last_column_in_users.rb:3]: Use 2 (not 1) spaces for indentation.

Comment (sumeetjain) [inline db/migrate/20160812144450_remove_first_last_column_in_users.rb:3]: Tab detected.

File: db/migrate/20160812144450_remove_first_last_column_in_users.rb [hunk 14]
```diff
@@ -0,0 +1,6 @@
...
+  	remove_column :users, :last_name, :string
```
Comment (sumeetjain) [inline db/migrate/20160812144450_remove_first_last_column_in_users.rb:4]: Tab detected.

File: db/schema.rb [hunk 15]
```diff
@@ -11,7 +11,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20160718193926) do
+ActiveRecord::Schema.define(version: 20160812144450) do
```
Comment (sumeetjain) [inline db/schema.rb]: Separate every 3 digits in the integer portion of a number with underscores(_).

File: db/seeds.rb [hunk 16]
```diff
@@ -15,8 +15,8 @@
   {award_name: 'Outstanding Youth Tech Award', relevant_fields: 1}])
 
 users = User.create([
-  {email: 'tommy@aol.com', password: 'hello314', first_name: 'Tommy', last_name: 'Tutone', weight: 3, admin: true},
-  {email: 'petericketts@netscape.com', password: 'isuckl0ts', first_name: 'Pete', last_name: 'Ricketts', weight: 1, admin: false}])
+  {email: 'tommy@aol.com', password: 'hello314', full_name: 'Tommy Tutone', weight: 3, admin: true},
```
Comment (sumeetjain) [inline db/seeds.rb:18]: Use 2 spaces for indentation in an array, relative to the first position after the preceding left parenthesis.

Comment (sumeetjain) [inline db/seeds.rb:18]: Line is too long. [100/80]

File: db/seeds.rb [hunk 17]
```diff
@@ -15,8 +15,8 @@
...
+  {email: 'petericketts@netscape.com', password: 'isuckl0ts', full_name: 'Pete Ricketts', weight: 1, admin: false}])
```
Comment (sumeetjain) [inline db/seeds.rb:19]: Line is too long. [116/80]

**Comment by halfghaninne** [comment]
👍  Good work! The wealth of comments above are automated code quality checks. I tried to delete the ones that are not relevant (migrations do not need descriptive comments, for example). These are things to keep in mind moving forward but don't really impact the merge-ability of your code. Nice work, again!

**Pull request merged**